orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+
+from unittest.mock import MagicMock, patch, call
+from typing import Generator, TypeVar, Optional
+from collections.abc import Iterable
+
+from hca_orchestration.contrib.argo_workflows import ExtendedArgoWorkflow, generate_argo_archived_workflows_client, \
+    ArgoFetchListOperation, ArgoArchivedWorkflowsClient
+from argo.workflows.client.models import V1alpha1Workflow, V1ObjectMeta, V1alpha1WorkflowSpec, V1alpha1Arguments, \
+    V1alpha1Parameter, \
+    V1alpha1WorkflowStatus
+
+from hca_orchestration.tests.support.mock_workflows import mock_argo_workflow
+
+
+def extend_workflow(workflow: V1alpha1Workflow) -> ExtendedArgoWorkflow:
+    return ExtendedArgoWorkflow(workflow, argo_url='https://nonexistentsite.test', access_token='token')
+
+
+class ArgoArchivedWorkflowsClientTestCase(unittest.TestCase):
+    def setUp(self):
+        self.client = ArgoArchivedWorkflowsClient("https://zombo.com", "tokentokentoken")
+
+    def workflows_to_page_of_results(self, archived_workflows: [], next_page: Optional[str]):
+        results_page = MagicMock()
+        results_page.items = archived_workflows
+        results_page.metadata._continue = next_page
+        return results_page
+
+    def test__pull_paginated_results_api_function_no_continue_results_one_call(self):
+        list_containing_3_V1alpha1Workflows = [
```

Comment:
this name is a tad redundant - usually i try to describe what i'm going to be using a variable for in its name, so here i'd use a name like `workflows_in_page`, and if there were multiple pages, maybe `workflows_in_page_one` and `workflows_in_page_two`.